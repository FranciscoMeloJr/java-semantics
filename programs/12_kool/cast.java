// Testing cast
// This program should execute but should not type check

class A {
  int x;

  A() {
    x=10;
  }

  int getA() {
    return x;
  }
}

class B extends A {
  int x;

  B() {
    super();
    x=20;
  }

  int getB() {
    return x;
  }
}

class main {
  main(String[] args) {
    B b = new B();
    A a = (A)b;
    System.out.println("b.x = "+ b.x);
    System.out.println("a.x = "+ a.x);
    System.out.println("a.getB() = "+ ((B)a).getB());
    System.out.println("a.getA() = "+ a.getA());
    System.out.println("Done!");
  }
}

public class cast {
  public static void main(String[] args) {
    new main(args);
  }
}

// b.x = 20
// a.x = 10
// a.getB() = 20
// a.getA() = 10
// Done!
