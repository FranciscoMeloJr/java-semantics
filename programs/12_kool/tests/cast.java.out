b.x = 0
a.x = 0
a.getB() = 0
a.getA() = 0
Done!
